lib/mu-query.cc

```diff
@@ -192,12 +190,12 @@ Query::Private::run_related (const std::string& expr, MuMsgFieldId sortfieldid,
         // moreover, in either threaded or non-threaded case, we sort the first
         // ("leader") query by date, i.e, we prefer the newest or oldest
         // (descending) messages.
-        const auto leader_qflags{QueryFlags::Leader};
+        const auto leader_qflags{QueryFlags::Leader | qflags};
         const auto threading{any_of(qflags & QueryFlags::Threading)};
 
         // Run our first, "leader" query
         DeciderInfo minfo{};
-        auto enq{make_enquire(expr, sortfieldid, leader_qflags)};
+        auto enq{make_enquire(expr, MU_MSG_FIELD_ID_DATE, leader_qflags)};
         const auto mset{enq.get_mset(0, maxnum, {},
                                      make_leader_decider(leader_qflags, minfo).get())};
 
```

Comment:
It just occurred to me... shouldn't we be gathering message IDs here, not thread IDs? Also, how are thread IDs populated in the db (as in, how are they generated, what actually are they) if threading is calculated after this point?

Reply:
Ah, I think I remember how this works now... there should be a 1:1 correspondence between message IDs and thread IDs, so this should work...